app/src/test/java/msi/paria/trainingapp/SampleViewModelTest.kt

```diff
@@ -0,0 +1,85 @@
+package msi.paria.trainingapp
+
+import io.bloco.faker.Faker
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+class SampleViewModelTest {
+    lateinit var sampleViewModel: SampleViewModel
+    lateinit var email : String
+    lateinit var faker : Faker
+
+    @Before
+    fun initValues() {
+        sampleViewModel = SampleViewModel()
+        faker = Faker()
+    }
+
+    @Test
+    fun test_email_isCorrect() {
+        email = faker.internet.email()
+        sampleViewModel.onEmailChanged(email)
+        sampleViewModel.onSubmitButtonClicked()
+        assertTrue(sampleViewModel.emailValidateState.value)
+    }
+
+    @Test
+    fun test_email_isWrong() {
```

Comment:
You can use a better function name for this test. for example: GivenWrongEmail_WhenEmaileChangedAndSubmitClicked_ThenValidationShouldBeFalse